<launch>
    <arg name="model" default="$(find hexapod)/urdf/hexapod.xacro"/>
    <arg name="servo_config" default="$(find hexapod)/yaml/servo_config.yaml"/>
    <arg name="ns" default="/"/>
    <arg name="gui" default="false"/>

    <group ns="$(arg ns)">        

        <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />

        <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
            <rosparam param="source_list">["test_joint_states"]</rosparam>
            <param name="use_gui" value="$(arg gui)"/>
        </node>
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

        <node pkg="hexapod" type="i2cpwm_controller" name="i2cpwm_controller" output="screen">
            <param name="i2c_dev_file" value="/dev/i2c-1" type="str" />
            <param name="config_file" value="$(arg servo_config)" />
        </node>
        
    </group>

</launch>